linkis-engineconn-plugins/engineconn-plugins/jdbc/src/main/scala/com/webank/wedatasphere/linkis/manager/engineplugin/jdbc/executer/JDBCEngineConnExecutor.scala

```diff
@@ -177,3 +190,17 @@ class JDBCEngineConnExecutor(override val outputPrintLimit: Int, val id: Int) ex
   }
 }
 
+
+object JDBCEngineConfig extends RPCMapCache[JobRequest, String, String](Configuration.CLOUD_CONSOLE_CONFIGURATION_SPRING_APPLICATION_NAME.getValue) {
+
+  override protected def createRequest(jobReq: JobRequest): CacheableProtocol = {
+    val engineTypeLabel = new EngineTypeLabel()
+    engineTypeLabel.setEngineType("jdbc")
+    engineTypeLabel.setVersion("4")
+    RequestQueryEngineTypeDefault(engineTypeLabel)
```

Comment:
This request can only get default configurations of jdbc engine, cannot get user configurations in governance station.
you can change to use RequestQueryEngineConfig. User and Creator labels can be found in EngineConnTask.

Reply:
![image](https://user-images.githubusercontent.com/14964489/130382315-6ea67cf8-5037-4eb7-ad0a-2053eda2d723.png)
I've found that the jdbc url is passed as engineconn-conf, but when executing the code the conf is not used. The conf is parsed in EngineConnServer, is there any way to take the conf in TaskExecutionServiceImpl.

By the way. I've found another bug. If the jdbc url contains char like & ,eg /test?characterEncoding=UTF-8&useSSL=false, it will make the command execute in the backgroud and ignore chars behind &. Then the engineConn start failed.

Reply:
> ![image](https://user-images.githubusercontent.com/14964489/130382315-6ea67cf8-5037-4eb7-ad0a-2053eda2d723.png)
> I've found that the jdbc url is passed as engineconn-conf, but when executing the code the conf is not used. The conf is parsed in EngineConnServer, is there any way to take the conf in TaskExecutionServiceImpl.
> 
> By the way. I've found another bug. If the jdbc url contains char like & ,eg /test?characterEncoding=UTF-8&useSSL=false, it will make the command execute in the backgroud and ignore chars behind &. Then the engineConn start failed.

Thank you very much for your careful research, I hope the following answers can help you.

1.The engine's default parameters are loaded during the engine startup, such as JDBC URL, UserName, Password, etc., this parameter is the default value in the configuration, rather than user-defined values, which are invalid. So we need to dynamically load the user's configuration during the task runtime, which is the real meaning of this Pull Request.

2.When the JDBC URL is dynamically obtained via the RPC, the & Symbol should not be lost. You can debug the JDBC engine and check the createDatasources method, which will make simple processing and storage of the URL, and then establish a JDBC connection directly.